test/parallel/test-http2-client-window-size.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+// This test ensures that servers are able to send data independent of window
+// size.
+
+const assert = require('assert');
+const common = require('../common');
+const h2 = require('http2');
+
+// Helper method which takes an array of buffers and returns a buffer containing
+// all the bytes in each buffer in order
+function concatBuffers(buffers) {
+  const resultLength =
+    buffers.reduce((prev, buf) => prev + buf.length, 0);
+  const result = Buffer.alloc(resultLength);
+  let offset = 0;
+  for (let i = 0; i < buffers.length; i++) {
+    buffers[i].copy(result, offset);
+    offset += buffers[i].length;
+  }
+  return result;
+}
```

Comment:
Since this is a test, you could also make use of `/internals/streams/BufferList.js` .. it's internal code but if you include `// Flags: --expose-internals` at the top of the test it will be accessible.